pkg/odo/cli/delete/component.go

```diff
@@ -91,11 +99,79 @@ func (o *ComponentOptions) deleteNamedComponent() error {
 	return nil
 }
 
-// deleteDevfileComponent deletes a component defined by the devfile in the current directory
+// deleteDevfileComponent deletes all the components defined by the devfile in the current directory
 func (o *ComponentOptions) deleteDevfileComponent() error {
+	devfileObj := o.EnvSpecificInfo.GetDevfileObj()
+	k8sComponents, err := libdevfile.ListKubernetesComponents(devfileObj, filepath.Dir(o.EnvSpecificInfo.GetDevfilePath()))
```

Comment:
Why are you not resuing `ListResourcesToDelete` (https://github.com/redhat-developer/odo/blob/daa9c61a4108dbef439842a17f993ebda74a8fcd/pkg/component/delete/delete.go#L26)?

Reply:
I'm not sure this is what we want to do here.

- With `ListResourcesToDelete`, we will find in the cluster all the resources belonging to a component, based on their labels.
- When deleting using a devfile, I understood that we want to delete the resources found in the devfile.

Reply:
@feloy that is a good point. I forgot about that. It makes sense to have two separate logics to handle this